docs/advanced_features.md

```diff
@@ -0,0 +1,71 @@
+# Advanced features
+
+## SPARQL+Text and SPARQL Autocompletion
+
+On top of the vanilla SPARQL functionality, QLever allows so-called SPARQL+Text
+queries on a text corpus linked to a knowledge base via entity recognition.  For
+example, the following query find all mentions of astronauts next to the words
+"moon" and "walk*" in the text corpus:
+
+    SELECT ?a TEXT(?t) SCORE(?t) WHERE {
+        ?a <is-a> <Astronaut> .
+        ?t ql:contains-entity ?a .
+        ?t ql:contains-word "walk* moon"
+    } ORDER BY DESC(SCORE(?t))
+
+Such queries can be simulated in standard SPARQL, but only with poor
+performance, see the CIKM'17 paper above.  Details about the required input data
+and the SPARQL+text query syntax and semantics can be found
+[here](docs/sparql_plus_text.md).
+
+QLever also supports efficient SPARQL autocompletion.  For example, the
+following query yields a list of all predicates associated with persons in the
+knowledge base, ordered by the number of persons which have that predicate.
+
+    SELECT ?predicate (COUNT(?predicate) as ?count) WHERE {
+      ?x <is-a> <Person> .
+      ?x ql:has-predicate ?predicate
+    }
+    GROUP BY ?predicate
+    ORDER BY DESC(?count)
+
+Note that this query could also be processed by standard SPARQL simply by
+replacing the second triple by ?x ?predicate ?object. However, that query is
```

Comment:
`replacying by` -> `replacuing with` that avoids having two repetitive bys. Also replace with is normally used in the active case (versus replace by in the passive case).